prog/minimization.go

```diff
@@ -121,11 +121,21 @@ func (typ *UnionType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bool
 }
 
 func (typ *PtrType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bool {
-	// TODO: try to remove optional ptrs
 	a := arg.(*PointerArg)
 	if a.Res == nil {
 		return false
 	}
+	if !ctx.triedPaths[path] {
+		r0 := a.Res
+		a.Res = nil
```

Comment:
This does not actually set pointer to NULL. This removes arg, but the address is still there. Change one of the tests to use an address different from 0x7f0000000000. I suspect this can even lead to crashes because `a.Res == nil && a.Address != 0` looks like a broken arg. Check out how MakePointerArg/MakeVmaPointerArg/MakeSpecialPointerArg setup pointer arg. You need to use `MakeSpecialPointerArg(typ, 0)` to create a proper NULL pointer.

Reply:
Done, added a test.